File: suitable/DigitalRoot/DigitalRoot.py
def find_max_digital_root(A, n):
    # 错误判断逻辑
    if n > 9 or n < -9 or (A <= 0 and n > 0) or (A < 0 and n >= 0):
        return "error_code"
    
    # n = 0 的特殊情况
    if n == 0 and A >= 0:
        return "0"
    
    # 从 A 开始向下寻找
    for i in range(A, -float('inf'), -1):
        if i == 0:
            digital_root = 0
        elif i > 0:
            digital_root = 1 + (i - 1) % 9
        else:
            digital_root = -(1 + (-i - 1) % 9)
        
        if digital_root == n:
            return str(i)
    
    return "没有找到符合条件的数字"

if __name__ == "__main__":
    A = int(input("请输入整数 A: "))
    n = int(input("请输入整数 n: "))
    
    result = find_max_digital_root(A, n)
    print("结果:", result)

File: suitable/DigitalRoot/DigitalRoot.java
import java.util.Scanner;

public class DigitalRoot {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        // 输入 A 和 n
        System.out.print("请输入整数 A: ");
        int A = scanner.nextInt();
        System.out.print("请输入整数 n: ");
        int n = scanner.nextInt();
        
        // 计算结果
        String result = findMaxDigitalRoot(A, n);
        System.out.println("结果: " + result);
    }

    private static String findMaxDigitalRoot(int A, int n) {
        // 错误判断逻辑
        if (n > 9 || n < -9 || (A <= 0 && n > 0) || (A < 0 && n >= 0)) {
            return "error_code";
        }
        
        // n = 0 的特殊情况
        if (n == 0 && A >= 0) {
            return "0";
        }
        
        // 从 A 开始向下寻找
        for (int i = A; i >= Integer.MIN_VALUE; i--) {
            int digitalRoot;
            if(i == 0) digitalRoot = 0;
            else if(i > 0) {
                digitalRoot = 1 + (i - 1) % 9;
            }
            else {
                digitalRoot = -(1 + (-i - 1) % 9);
            }
            if (digitalRoot == n) {
                return String.valueOf(i);
            }
        }
        
        return "没有找到符合条件的数字";
    }
}